pwa-devdocs/src/tutorials/hello-upward/adding-react/index.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Adding React
+---
+
+This tutorial teaches you how to add webpack to your UPWARD project and use it to create a React application.
```

Comment:
I think this topic can be made much simpler by getting rid of all the "extras" around React.

Since [React doesn't actually need JSX](https://reactjs.org/docs/react-without-jsx.html) you can get rid of the dependency on `babel` and `webpack` here.

Check it out:

## `hello-world.mst`

```jsx
{{> templates/open-document}}

<title>{{ title }}</title>
<script src="https://cdnjs.cloudflare.com/ajax/libs/react/16.8.6/umd/react.production.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/16.8.6/umd/react-dom.production.min.js"></script>

{{> templates/open-body}}

<div id="root">Loading...</div>

<script defer src="/app.js"></script>

{{> templates/close-document}}
```

Notice that the template is pulling in React and React DOM from CDNs. This eliminates the need to install them locally.

## `app.js`

```js
const App = () => {
    return React.createElement(
        'h1',
        null,
        'Hello World from React!'
    );
};

ReactDOM.render(
    React.createElement(App),
    document.getElementById('root')
);
```

Notice that we aren't using JSX here and instead using the `React.createElement` function directly. This is [what Babel does](https://babeljs.io/en/repl) for you.

The only thing left to do is fix up the UPWARD spec a bit:

```yaml
status: response.status
headers: response.headers
body: response.body

response:
  resolver: conditional
  when:
      - matches: request.url.pathname
        pattern: '^/?$'
        use: helloWorld
      - matches: request.url.pathname
        pattern: '^/hello-world/?$'
        use: helloWorld
      - matches: request.url.pathname
        pattern: '^/app.js$'
        use: appScript
  default: notFound

helloWorld:
  inline:
      status:
          resolver: inline
          inline: 200
      headers:
          resolver: inline
          inline:
              content-type:
                  resolver: inline
                  inline: 'text/html'
      body:
          resolver: template
          engine: mustache
          template: './templates/hello-world.mst'
          provide:
            title:
              resolver: inline
              inline: 'This is the page title!'
notFound:
    inline:
        status:
            resolver: inline
            inline: 404
        headers:
            resolver: inline
            inline:
                content-type:
                    resolver: inline
                    inline: 'text/string'
        body:
            resolver: inline
            inline: 'Page not found!'

appScript:
  inline:
    status:
      resolver: inline
      inline: 200
    headers:
      resolver: inline
      inline:
        content-type:
          resolver: inline
          inline: 'application/javascript'
    body:
      resolver: file
      file:
        resolver: inline
        inline: './app.js'
      encoding:
        resolver: inline
        inline: 'utf-8'
```
<hr />

Since this is an UPWARD tutorial, I think this greatly simplifies the topic and introduces readers to the `FileResolver` more quickly.

We probably want to include a paragraph about how production code should use webpack and babel but I don't think they need to be introduced in this tutorial.

Reply:
Even though it's billed as an UPWARD tutorial, I also wanted the end result of doing this tutorial to be an environment that the reader can use as a starting point for their PWA Studio projects. This is something the community has been asking us for.

It's true that React doesn't require JSX, but most examples out there use JSX and ES6 syntax. I think it would be a bad experience for developers if we introduce how something is done in the tutorial and promote something very different in the implementation code.